### PR TITLE
Merge v0.8 from mvp_demo to master

### PR DESCRIPTION
**Pull request by chandrams**
## Summary by Sourcery

Update accelerator support, bulk API validation, performance profile lifecycle, term settings, and test/CI tooling for the 0.8 release.

New Features:
- Add accelerator profiles, recommendation mappings, and metadata for new NVIDIA GPUs including H200, B200, H100 94/96GB, and RTX Pro Blackwell cards.
- Extend bulk API with request/time-range/datasource validation and corresponding error handling.
- Introduce configurable term definitions and defaults for recommendation engines, including calendar-based and duration-based terms.
- Add create, list, update, and delete performance profile API coverage with new negative and lifecycle tests.
- Add accelerator metrics support to monitoring mode APIs and document accelerator-aware recommendations.

Bug Fixes:
- Ensure unsupported accelerators are logged and surfaced via notifications instead of silently ignored.
- Fix performance profile SLO equality checks and mandatory field validation messages.
- Harden bulk API behavior by fixing in-memory job-cache flag usage and removing stack trace printing in error paths.
- Prevent deletePerformanceProfile from removing profiles that are still associated with experiments.
- Fix pod selection in common utils when optional ignore filters are unset and improve metric-profile listing when DB is empty.

Enhancements:
- Refine accelerator model detection, framebuffer calculation, and profile lookup logic to support additional GPUs and MIG partitions.
- Simplify MIG profile-to-recommendation mapping using a switch construct and centralize accelerator name/memory constants and logs.
- Add accelerator-related notification codes and wire them into recommendation flows and test expectations.
- Add term settings container and defaults for duration-based recommendation terms, including plotting parameters.
- Refine performance profile service list handling and cleanup unused DAO usage in experiment creation.

Build:
- Bump project version to 0.8 and refresh dependencies including Fabric8, AWS SDK, PubNub, and Kafka clients.
- Upgrade Java target to version 25 and update Prometheus installation script to a newer kube-prometheus release and CRD-first apply flow.
- Update Docker image tags and manifests to use autotune_operator:0.8 and a new quay.io test image namespace.

CI:
- Update GitHub Actions workflows to newer Minikube and Kubernetes versions and to use the new test Docker image name in push/deploy jobs.

Deployment:
- Adjust CRC deployment manifests (minikube, OpenShift, AKS, BYODB) to point to the 0.8 operator image and tune resource requests for scale tests.

Documentation:
- Document accelerator metric support, supported NVIDIA GPUs, and MIG recommendation formats in a new AcceleratorSupport guide.
- Extend MonitoringModeAPI examples with accelerator metrics payloads and reference accelerator documentation.
- Document deletePerformanceProfile API and expand performance profile API documentation.
- Add detailed notification code ranges for accelerator subsystems and new notification codes.
- Add formal test plans for releases 0.7.1 and 0.8 and update scalability and test-suite documentation for new scenarios.

Tests:
- Extend bulk API local monitoring tests for time-range validation and missing datasource handling.
- Broaden remote monitoring tests to cover multiple GPU models, accelerator notifications, and GPU scalability scenarios.
- Add comprehensive tests for performance profile create/list/update/delete APIs including duplicate, missing-field, and association cases.
- Refactor scalability tests to support GPU container experiments and parameterize workloads, and update scale/migration scripts accordingly.
- Tighten recommendation JSON schema validation and job status expectations in helper utilities.

### CODE REVIEW

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_performance_profiles.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    response = list_performance_profiles()
+    data = response.json()
+    print("response = ", data)
+    assert response.status_code == SUCCESS_200_STATUS_CODE
+    # Assert the response is an empty list
+    assert isinstance(data, list)
+    assert len(data) == 0
+
+
+@pytest.mark.perf_profile
```
[the remark this answers is not in the document]
Comment (khansaad) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_list_performance_profiles.py:54]: @sourcery-ai Its just re-using the `data` variable again to hold the response message

File: tests/scripts/remote_monitoring_tests/rest_apis/test_delete_performance_profile.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    ("objective_function", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("function_type", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("function_variables", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("metric_name", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("datasource", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("value_type", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("aggregation_functions", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("function", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("query", ERROR_STATUS_CODE, ERROR_STATUS)
+]
```
[the remark this answers is not in the document]
Comment (khansaad) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_delete_performance_profile.py:40]: Yes, this is unused. Will be removed later

File: tests/scripts/local_monitoring_tests/rest_apis/test_bulkAPI.py [hunk 3]
```diff
@@ -104,3 +99,125 @@ def test_bulk_post_request(cluster_type, bulk_request_payload, expected_job_id_p
...
+@pytest.mark.parametrize("start, end, expected_error", [
+    ("2025-01-01T12:00:00Z", "2025-01-02T12:00:00Z", "Valid"),               # Valid scenario
+    ("", "", "Invalid date format. Must follow ISO 8601 format (YYYY-MM-DDTHH:mm:ss.sssZ) for the jobId:"), # empty
+    ("2024-01-01 10:00:00", "2024-01-01T12:00:00Z", "Invalid date format. Must follow ISO 8601 format (YYYY-MM-DDTHH:mm:ss.sssZ) for the jobId:"),  # bad format
+    ("2025-01-02T12:00:00Z", "2025-01-01T12:00:00Z", "Start time should be before end time for the jobId:"),  # start > end
+    ])
+def test_bulk_api_time_range_validation(cluster_type, start, end, expected_error, caplog):
+    """
+    Validates all negative time-range scenarios for Bulk API.
+    """
```
[the remark this answers is not in the document]
Comment (khansaad) [reply tests/scripts/local_monitoring_tests/rest_apis/test_bulkAPI.py:113]: The tests were added considering the scope of the JIRA ticket. We can increase the test coverage in the future release.

File: src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java [hunk 4]
```diff
@@ -62,9 +62,209 @@ private static void initializeAcceleratorProfiles() {
...
+        h100_96_gb_profiles.add(new AcceleratorProfile(
+                AnalyzerConstants.AcceleratorConstants.AcceleratorProfiles.H100_PROFILE_7G_96GB,
+                1,
+                1,
+                1
+        ));
+
+
+        acceleratorProfilesMap.put(AnalyzerConstants.AcceleratorConstants.SupportedAccelerators.A100_80_GB, commonProfiles);
+        acceleratorProfilesMap.put(AnalyzerConstants.AcceleratorConstants.SupportedAccelerators.H100_80_GB, commonProfiles);
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java:260]: @sourcery-ai I have added `commonProfiles` intentionally as it's a read only structure and if we call `new` arraylist we will be maintaining two copies of same readonly structures, to save memory, I have intentionally used a common profile for `A100 80GB` and `H100 80GB` instances as they share same profiles.

[the remark this answers is not in the document]
Comment (bharathappali) [reply src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java:260]: > if they are guaranteed immutable, consider documenting that assumption here.

Yes they are guaranteed immutable. Will add comments for the same.

File: design/AcceleratorSupport.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      },
+      "nvidia.com/mig-4g.20gb": {
+        "amount": 1.0,
+        "format": "cores"
+      }
+    }
+}
+```
+
+#### MIG Supported Nvidia GPU's which can be optimised by KRUIZE
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/AcceleratorSupport.md:146]: @sourcery-ai Thanks for review will fix it next PR

File: design/AcceleratorSupport.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+| ----------------------------- |-------------------------------------------------------| --------------------------- | ------------- |
+| `acceleratorCoreUsage`        | GPU core utilization over the reporting interval      | `DCGM_FI_DEV_GPU_UTIL`      | Percentage    |
+| `acceleratorMemoryUsage`      | GPU memory copy / memory controller utilization       | `DCGM_FI_DEV_MEM_COPY_UTIL` | Percentage    |
+| `acceleratorFrameBufferUsage` | GPU frame buffer (absolute device memory) consumption | `DCGM_FI_DEV_FB_USED`       | MiB           |
+
+Each accelerator metric is reported as aggregated values (for example, min, max, avg) over the specified time interval. Metadata such as the GPU model needs to be provided to help Kruize contextualize and calculate the partition required for the workload based on the model it's using.
+
+NOTE: Accelerator MIG recommendations are not similar to CPU and Memory which are common across the hardware, GPU recommendations provided by kruize are specific MIG resource profiles that NVIDIA has developed specific to a particular GPU.
+Please refer https://docs.nvidia.com/datacenter/tesla/mig-user-guide/supported-mig-profiles.html for more information.
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/AcceleratorSupport.md:54]: @sourcery-ai Thanks for review will fix it next PR

File: design/AcceleratorSupport.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+**Supported Architectures:** `Ampere`, `Hopper`, `Blackwell`
+
+## MODE: Remote Monitoring
+
+### Overview
+
+Kruize supports a Remote Monitoring mode that enables users to create experiments for workloads running on external or remote Kubernetes clusters. In this mode, Kruize does not directly scrape metrics; instead, users periodically push aggregated resource usage data to Kruize using the Update Results API. Based on this data, Kruize generates resource optimization recommendations.
+
+Initially, Remote Monitoring supported only CPU and memory metrics. This capability has now been extended to include accelerator (GPU) metrics, enabling GPU-aware monitoring and recommendations for supported NVIDIA GPUs [Please check the list mentioned in supported GPU's section].
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/AcceleratorSupport.md:20]: @sourcery-ai Thanks for review will fix it next PR

File: design/PerformanceProfileAPI.md [hunk 8]
```diff
@@ -761,3 +760,40 @@ This is quick guide instructions to update performance profile using input JSON
 }
 ```
 #### Note: One of query or aggregation_functions is mandatory. Both can be present together.
+
+
+## DeletePerformanceProfile
+
+This is quick guide instructions to delete performance profile using input param as follows.
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/PerformanceProfileAPI.md:767]: @sourcery-ai Thanks for review will fix it next PR

File: design/NotificationCodes.md [hunk 9]
```diff
@@ -2,87 +2,91 @@
...
+| 324002 |  NOTICE  |                MEMORY_RECORDS_NOT_AVAILABLE                |             Specifies that the Memory recordings are NOT AVAILABLE              |               Memory metrics are not available, No Memory Recommendations can be generated               | DATA USER |
+| 324003 |  NOTICE  |                 MEMORY_REQUESTS_OPTIMISED                  |          Specifies that the workload is optimised wrt MEMORY REQUESTS           |                       Workload is optimised wrt MEMORY REQUESTS, no changes needed                       | DATA USER |
+| 324004 |  NOTICE  |                  MEMORY_LIMITS_OPTIMISED                   |           Specifies that the workload is optimised wrt MEMORY LIMITS            |                        Workload is optimised wrt MEMORY LIMITS, no changes needed                        | DATA USER |
+| 328001 |  NOTICE  |                 ACCELERATOR_NOT_SUPPORTED                  |       Specifies that the accelerator provided is not supported by kruize        |                                  Accelerator is not supported by kruize                                  | DATA USER |
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/NotificationCodes.md:85]: @sourcery-ai Thanks for review will fix it next PR

File: tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py [hunk 10]
```diff
@@ -816,7 +816,8 @@ def test_update_namespace_recommendations_for_diff_reco_terms_with_only_latest(t
 # Test for GPU recommendations
 # This test checks if kruize provides GPU recommendations for valid GPU's
 @pytest.mark.sanity
-def test_update_valid_accelerator_recommendations(cluster_type):
+@pytest.mark.parametrize("gpu_name", SUPPORTED_GPUS)
+def test_update_valid_accelerator_recommendations(cluster_type, gpu_name):
     '''
         Creates Experiment +
         update results for GPU workload with 2 records
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py:823]: @sourcery-ai Thanks for review will fix it next PR
